source/Public/New-GuestConfigurationPolicy.ps1

```diff
@@ -275,26 +296,39 @@ function New-GuestConfigurationPolicy
 
     # Download package
     $tempPath = Reset-GCWorkerTempDirectory
+    $packagePath = Join-Path -Path $tempPath -ChildPath 'extracted'
 
-    $packageFileDownloadName = 'temp.zip'
-    $packageFileDownloadPath = Join-Path -Path $tempPath -ChildPath $packageFileDownloadName
-
-    if (Test-Path -Path $packageFileDownloadPath)
+    if ($ContentManagedIdentity -and $ContentPath)
     {
-        $null = Remove-Item -Path $packageFileDownloadPath -Force
-    }
+        $contentHash = (Get-FileHash -Path $ContentPath -Algorithm 'SHA256').Hash
 
-    $null = Invoke-WebRequest -Uri $ContentUri -OutFile $packageFileDownloadPath
+        $packagePresentIn = $ContentPath
 
-    if ($null -eq (Get-Command -Name 'Get-FileHash' -ErrorAction 'SilentlyContinue'))
+        Write-Information -MessageData "Arc is not supported for use with User Assigned Idenity" -InformationAction Continue
+    }
+    else
     {
-        $null = Import-Module -Name 'Microsoft.PowerShell.Utility'
+        $packageFileDownloadName = 'temp.zip'
+        $packageFileDownloadPath = Join-Path -Path $tempPath -ChildPath $packageFileDownloadName
+
+        if (Test-Path -Path $packageFileDownloadPath)
+        {
+            $null = Remove-Item -Path $packageFileDownloadPath -Force
+        }
+
+        $null = Invoke-WebRequest -Uri $ContentUri -OutFile $packageFileDownloadPath
+
+        if ($null -eq (Get-Command -Name 'Get-FileHash' -ErrorAction 'SilentlyContinue'))
+        {
+            $null = Import-Module -Name 'Microsoft.PowerShell.Utility'
+        }
+        $contentHash = (Get-FileHash -Path $packageFileDownloadPath -Algorithm 'SHA256').Hash
+
+        $packagePresentIn = $packageFileDownloadPath
```

Comment:
No need for $packagePresentIn.
You can just use $packageFileDownloadPath

Reply:
Changed the code accordingly.